python/ray/ml/checkpoint.py

```diff
@@ -332,12 +344,47 @@ def to_directory(self, path: Optional[str] = None, dedup: bool = True) -> str:
                     shutil.copytree(local_path, path)
             elif external_path:
                 # If this exists on external storage (e.g. cloud), download
-                download_from_uri(uri=external_path, local_path=path)
+                download_from_uri(uri=external_path, local_path=path, filelock=False)
             else:
                 raise RuntimeError(
                     f"No valid location found for checkpoint {self}: {self._uri}"
                 )
 
+    def to_directory(self, path: Optional[str] = None) -> str:
+        """Write checkpoint data to directory.
+
+        Args:
+            path: Target directory to restore data in. If not specified,
+                will create a temporary directory. The name of the temporary
+                directory will be random, unless the checkpoint was created
+                from an object reference, in which case it will be equal
+                to the ID of that object reference.
+
+        Returns:
+            str: Directory containing checkpoint data.
+        """
+        user_provided_path = path is not None
+        path = path if has_path else self._get_temporary_checkpoint_dir()
+
+        _make_dir(path, acquire_del_lock=not user_provided_path)
+
+        try:
+            # Timeout 0 means there will be only one attempt to acquire
+            # the file lock. If it cannot be aquired, a TimeoutError
+            # will be thrown.
+            with FileLock(f"{path}.lock", timeout=0):
```

Comment:
What happens if I have a trailing slash, e.g. `checkpoint.to_directory("/some/path/")`?

Reply:
taken care of